### PR TITLE
Live audit data

### PR DESCRIPTION
**Pull request by djarotech**
this PR is small. I changed the rc.yaml so that I can do live data . Tested on a gcp instance

### CODE REVIEW

**Comment by devdattakulkarni** [comment]
Looks fine. Couple of points:

1) Can you squash all the changes in a single commit

2) In the commit message add more context about the change. Have the commit header line a crisp one-line description of change. Then leave a blank line. In the body of the commit message add short description of what the change is, what is its purpose, if there are any special instructions to enable it.

3) I see that you have tested it on GCP. So in a real cluster this will work. What about Minikube though? That will be the first place anyone want to try. Can you investigate bit around what would it take to use live data on Minikube?

File: artifacts/simple-image/Dockerfile [hunk 1]
```diff
@@ -14,4 +14,8 @@
 
 FROM fedora
 ADD kube-provenance-apiserver /
+<<<<<<< HEAD
```
Comment (devdattakulkarni) [inline artifacts/simple-image/Dockerfile]: Stray line coming probably coming from merge conflict.
Should be removed.

File: artifacts/simple-image/Dockerfile [hunk 2]
```diff
@@ -14,4 +14,8 @@
...
+ADD minikube-sample-audit.log /tmp/
+=======
+>>>>>>> 258a45ce55e9cde9247d86289d08bb80090ca50f
```
Comment (devdattakulkarni) [inline artifacts/simple-image/Dockerfile]: Same comment as above for lines 19 and 20.

File: artifacts/example/rc.yaml [hunk 3]
```diff
@@ -37,5 +41,4 @@ spec:
         - name: audit-log
           hostPath:
             path: /tmp/kube-apiserver-audit.log
-            # this field is optional
-            type: File
+            type: FileOrCreate
```
Comment (devdattakulkarni) [inline artifacts/example/rc.yaml]: +1

File: pkg/provenance/provenance.go [hunk 4]
```diff
@@ -73,14 +75,30 @@ func init() {
 	AllProvenanceObjects = make([]ProvenanceOfObject, 0)
 
 }
-
+func onMinikube() bool {
+	ip := os.Getenv("HOST_IP")
+	ipBeginsWith := strings.Split(ip, ".")[0]
+	return ipBeginsWith == "10"
```
Comment (devdattakulkarni) [inline pkg/provenance/provenance.go]: Minikube IPs also begin with 198. for v0.25 and below.
That should be accommodated here as well.
You can have a OR condition for these two cases.

**Comment by devdattakulkarni** [comment]
Check my in-line comments.

Also, two other comments.

1) Can you squash all the commits into a single commit? Check how to do it here:

https://stackoverflow.com/questions/14534397/squash-all-my-commits-into-one-for-github-pull-request

https://gist.github.com/patik/b8a9dc5cd356f9f6f980

Once you have squashed the commits you will have to force push the branch as shown on the second link.

2) Make the commit message informative. Explain about two options being supported (Minikube and real setup). Point out that on Minikube live data is not being used.

**Comment by djarotech** [comment]
Done! please review

File: pkg/provenance/provenance.go [hunk 5]
```diff
@@ -420,59 +429,63 @@ func handleCompositeFields(vSliceMap []map[string]string, mapRelationships map[s
 	}
 	return false
 }
+//Method that returns true if all the elements in boolSlice are True
+func all(boolSlice []bool) bool{
```
Comment (devdattakulkarni) [inline pkg/provenance/provenance.go]: This change seems not related to Audit logs.

**Comment by devdattakulkarni** [comment]
Looks good after squashing. Few more points related to commit message:

Top line: Use following
"Support for live audit data (and misc. changes)"

Body: Include following two points:
- Current top-line + body (combine them)
- Miscellaneous formatting changes.

File: pkg/provenance/provenance.go (hunk 5 above)
Comment (djarotech) [reply pkg/provenance/provenance.go]: How strange . This method was merged in the last pull request, the left hand side (master) looks incorrect  .
